edweiss-app/__test__/showtime/LectureScreen.test.tsx

```diff
@@ -76,11 +76,13 @@ jest.mock('@/hooks/useListenToMessages', () => jest.fn());
 jest.mock('expo-router', () => ({
     router: { push: jest.fn() },
     Stack: {
-        Screen: jest.fn(({ options }) => (
-            <>{options.title}</>
-        )),
+        Screen: jest.fn(({ options }) => <>{options.title}</>),
     },
-    useLocalSearchParams: jest.fn(() => ({ courseNameString: 'testCourse', lectureIdString: 'testLectureId' }))
+    useLocalSearchParams: jest.fn(() => ({
+        courseNameString: 'testCourse',
+        lectureIdString: 'testLectureId',
+    })),
+    useFocusEffect: jest.fn(),
```

Comment:
I see in the new codebase that you use the `useFocusEffect`, can you explain me what it's does. 
I am still a dummy at react, but can you explain me also why did you choose to use it. Thanks

Reply:
I have used this useFocusEffect for an action on callback to be detected. So in other words I wanted to detect when the user would do a router.back and do some lines of code before the user actually left the screen. I had this in mind because I sometimes used ScreenOrientation lock to maintain the user in a certain display mode and if they did a router.back in that specific moment all screens in the app would stay locked in that orientation. 
So with this function I can always maintain it unlocked by just unlocking before actually changing to the previous screen

This can also be useful for you as well if you happen to use ScreenOrientation.lockAsync

For why I chose this specific function I just used it because of the ones I tested it was the only one that actually managed to handle the effect that I wanted.

Reply:
Oh that's really great. I did not knew we could do it this way. Thank's for sharing it. Defenetly that's a great usecase.